endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1028,40 +1028,50 @@ func TestAMPSiteExt(t *testing.T) {
 
 // TestBadRequests makes sure we return 400's on bad requests.
 func TestAmpBadRequests(t *testing.T) {
-	dir := "sample-requests/invalid-whole"
+	dir := "sample-requests/amp/invalid-whole"
 	files, err := os.ReadDir(dir)
 	assert.NoError(t, err, "Failed to read folder: %s", dir)
 
-	badRequests := make(map[string]json.RawMessage, len(files))
 	for index, file := range files {
-		badRequests[strconv.Itoa(100+index)] = readFile(t, "sample-requests/invalid-whole/"+file.Name())
-	}
+		filename := file.Name()
+		fileData := readFile(t, "sample-requests/amp/invalid-whole/"+filename)
 
-	endpoint, _ := NewAmpEndpoint(
-		fakeUUIDGenerator{},
-		&mockAmpExchange{},
-		newParamsValidator(t),
-		&mockAmpStoredReqFetcher{badRequests},
-		empty_fetcher.EmptyFetcher{},
-		&config.Configuration{MaxRequestSize: maxSize},
-		&metricsConfig.NilMetricsEngine{},
-		analyticsBuild.New(&config.Analytics{}),
-		map[string]string{},
-		[]byte{},
-		openrtb_ext.BuildBidderMap(),
-		empty_fetcher.EmptyFetcher{},
-		hooks.EmptyPlanBuilder{},
-		nil,
-	)
-	for requestID := range badRequests {
-		request := httptest.NewRequest("GET", fmt.Sprintf("/openrtb2/auction/amp?tag_id=%s", requestID), nil)
+		test, err := parseTestData(fileData, filename)
+		if !assert.NoError(t, err) {
+			return
+		}
+
+		requestID := strconv.Itoa(100 + index)
+
+		endpoint, _ := NewAmpEndpoint(
```

Comment:
When it comes to testing, CPU time is not that important but, instantiating the endpoint at every turn inside the `for` loop seems a bit wasteful. Can we move the endpoint instantiation outside of the `for` loop?